packages/proxy/src/agent.ts

```diff
@@ -47,7 +47,7 @@ class CustomAgents {
   }
 
   private getParsedUrl() {
-    return this.proxy ? new URL(this.proxy) : new URL(this.url);
+    return new URL(this.url);
```

Comment:
I don't think this is good practice, the https_proxy variable is intended to specify a proxy server for HTTPS traffic, should not rely on the registry protocol and might damage current implementations.

Reply:
Ok, get it. Can we detect protocol from `http_proxy, https_proxy` config key names? i.e. use `http_proxy` config to construct a `HttpProxyAgent` and `https_proxy` config to `HttpsProxyAgent`.

Reply:
That's how it's implied in the [docs](https://verdaccio.org/docs/next/configuration/#http_proxy-and-https_proxy). Other combinations like the one you mentioned in the initial comment should lead to an error or a warning at startup of Verdaccio.